packages/cli/src/lib/cache/SuccessCache.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs-extra';
+import { resolve as resolvePath } from 'node:path';
+
+const DEFAULT_CACHE_BASE_PATH = 'node_modules/.cache/backstage-cli';
+
+const CACHE_MAX_AGE_MS = 7 * 24 * 3600_000;
+
+export class SuccessCache {
+  readonly #path: string;
+
+  constructor(name: string, basePath?: string) {
+    this.#path = resolvePath(basePath ?? DEFAULT_CACHE_BASE_PATH, name);
+  }
+
+  async read(): Promise<Set<string>> {
+    const state = await fs.stat(this.#path).catch(error => {
+      if (error.code === 'ENOENT') {
+        return undefined;
+      }
+      throw error;
+    });
+    if (!state || !state.isDirectory()) {
```

Comment:
you're permissive with non-directories here, but does `fs.ensureDir` handle it gracefully in `write()` below or will it throw if it finds a file?

Reply:
Nah it errors if there's a file already. Tbh this bit could error on that too, just felt ugly to call readDir of we know it's not a dir x)